scripts/dtsBundler.mjs

```diff
@@ -310,6 +318,10 @@ function verifyMatchingSymbols(decl) {
             }
             const symbolInScope = findInScope(symbolOfNode.name);
             if (!symbolInScope) {
+                if (symbolOfNode.declarations?.every(d => isLocalDeclaration(d) && d.getSourceFile() === decl.getSourceFile()) && !isSelfReference(node, symbolOfNode)) {
+                    // The symbol is a local that needs to be copied into the scope.
+                    scopeStack[scopeStack.length - 1].locals.set(symbolOfNode.name, { symbol: symbolOfNode, writeTarget: isInternal ? WriteTarget.Internal : WriteTarget.Both });

```

Comment:
It seems moderately weird for this verification function to modify the locals, but not sure what's better.

Reply:
I feel like I considered either renaming the function or returning the set of new locals to be appended before I paused this work.

Reply:
@jakebailey any more thoughts on this? I agree this code isn’t great. Otherwise I think the current state is ready for review.